### PR TITLE
Add a TLV encoder/decoder.

### PR DESCRIPTION
**Pull request by zhenlu**

### CODE REVIEW

File: uma/test/tlv_utils_test.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+	return b.Data, nil
+}
+
+func (b *BinaryCodableStruct) UnmarshalBytes(data []byte) error {
+	b.Data = data
+	return nil
+}
+
+type TLVUtilsTests struct {
+	StringField string `tlv:"0"`
```
Comment (jklein24) [inline uma/test/tlv_utils_test.go:23]: This is pretty cool :-D

File: uma/utils/tlv_utils.go [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+		return nil, fmt.Errorf("marshal requires a pointer to a struct")
+	}
+
+	val = reflect.Indirect(val)
+	typ := val.Type()
+
+	var result []byte
+	for i := 0; i < val.NumField(); i++ {
+		field := val.Field(i)
+		tag := typ.Field(i).Tag.Get("tlv")
```
Comment (jklein24) [inline uma/utils/tlv_utils.go:36]: This is cool! Is this how the json serializing works in Go too?

File: uma/utils/tlv_utils.go [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+		if tag == "" {
+			continue
```
Comment (jklein24) [inline uma/utils/tlv_utils.go:38]: Will this just silently drop those fields? Do you think it should it be an error?

**Review by jklein24 — APPROVED** [review]

File: uma/utils/tlv_utils.go (hunk 2 above)
Comment (zhenlu) [reply uma/utils/tlv_utils.go:36]: Yeah I think json utilize this reflect as well

File: uma/utils/tlv_utils.go (hunk 3 above)
Comment (zhenlu) [reply uma/utils/tlv_utils.go:38]: The consideration is to make it flexible, so not all fields needs to be serialized to TLV.

Comment (zhenlu) [reply uma/utils/tlv_utils.go:38]: One example is that if we ever want to use 2 bytes as type, we can specify it in the struct without TLV field.

**Pull request merged**